src/registry/currency/edit-validation.ts

```diff
@@ -0,0 +1,36 @@
+import {IntlShape} from 'react-intl';
+import {z} from 'zod';
+
+import {buildCommonSchema} from '@/registry/validation';
+
+// undefined (optional) for unspecified, otherwise a finite numeric value. Note that
+// null would be nicer, but formio's schema does not support null for validate.min,
+// validate.max or defaultValue
+const currencySchema = z.number().finite().optional();
+
+// case for when component.multiple=false
+const singleValueSchema = z
+  .object({multiple: z.literal(false)})
+  .and(z.object({defaultValue: currencySchema}));
+
+// case for when component.multiple=true
+const multipleValueSchema = z
+  .object({multiple: z.literal(true)})
+  .and(z.object({defaultValue: currencySchema.array()}));
+
+const defaultValueSchema = singleValueSchema.or(multipleValueSchema);
```

Comment:
Should currency also be only single capable, like number?

Reply:
yeah let's drop the "multiple" behaviour. I can't make sense of it.

It would be wise to write a detection script in the backend so that we check if any (prod) instances make use of it.